src/functions-and-arrays.js

```diff
@@ -1,41 +1,137 @@
+  else if (numberA < numberB) return numberB;
+  else return numberA;
+}
+//console.log(maxOfTwoNumbers(5, 5))
 
 
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
-
+function findLongestWord(stringsArray) {
+  if (stringsArray.length === 0) return null;
+  else if (stringsArray.length === 1) return stringsArray[0];
+  else {
+    // check for the longest word
+      let charCount = 0;
+      let longest = null;
+      for (let i = 0; i < stringsArray.length; i++)
+      if (stringsArray[i].length > charCount) {
+      charCount = stringsArray[i].length;
+      longest = stringsArray[i];
+  }
+  return longest;
+}
+}
+findLongestWord(words);
 
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
-
 
+function sumNumbers(numsArray) {
+  if (!numsArray.length) return 0;
+  let sum = 0;
+  numsArray.forEach(element => {
+    sum += element;
+    return sum;
+  });
+  return sum;
+}
 
-// Iteration #3.1 Bonus:
-function sum() {}
 
 
+// Iteration #3.1 Bonus:
+function sum(array) {
+
+  if (!array.length) return 0; // empty array return 0
+
+  let result = 0; 
+  for (let i = 0; i < array.length; i++) {
+    // Loop through each element to check the data type and assign values:
+    // 1. Boolean:
+      if(typeof array[i] === "boolean") { 
+        // if boolean is true:
+        if(array[i] === true) {
+          result ++;
+        // else if boolean is false:
+        }else if(array[i] === false) {
+          result += 0;
+        }
+    // 2. Number:
+      } else if (typeof array[i] === "number") {
+        result += array[i];
+    // 3. String:
+      } else if (typeof array[i] === "string") {
+        result += array[i].length;
+    // 4. 
+      } else if (typeof array[i] === "array" || (typeof array[i] === "object")){
+        throw new Error("Unsupported data type sir or ma'am");
+      }
+  }
+  return result;
+}
 
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
+
+
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
-function averageNumbers() {}
+function averageNumbers(numsArray){
+  if(!numsArray.length) return null
+
+  let sum = 0;
+  for (let i = 0; i < numsArray.length; i++) {
+    sum += numsArray[i];
+  }
+  return sum / numsArray.length
+}
+
 
 
 // Level 2: Array of strings
 const wordsArr = ['seat', 'correspond', 'linen', 'motif', 'hole', 'smell', 'smart', 'chaos', 'fuel', 'palace'];
 
-function averageWordLength() { }
+function averageWordLength(arrayOfStrings) {
+  if (!arrayOfStrings.length) return null;
+
+  let sum = 0
+
+  for (let i = 0; i < arrayOfStrings.length; i++) {
+    sum += arrayOfStrings[i].length;
+  }
+  return sum / arrayOfStrings.length;
+}
 
 // Bonus - Iteration #4.1
-function avg() {}
+// Same as Iteration 3.1 but without the error and with average.
+function avg(array) {
+  if(array == 0) return null;
```

Comment:
Always use '===' instead of '=='. And here, you were checking for the actual 0 and not an empty array. You forgot the length. `if(array.length === 0) return null;`